src/object_store/src/object/mod.rs

```diff
@@ -335,7 +352,9 @@ impl<OS: ObjectStore> MonitoredObjectStore<OS> {
             .operation_latency
             .with_label_values(&["read"])
             .start_timer();
-
+        if let StoreMediaTypeE::S3 = self.inner.store_media_type() {
```

Comment:
Shall we can use a function like `ReportWrite` or `ReportRead` to shadow the type match logic, and use it for other `StoreMediaType` uniformly ?

Reply:
Good point.